common/src/test/resources/testPayment.properties

```diff
@@ -0,0 +1,10 @@
+# needed for convinience to make parsing easier and more independed of parameters from outside
+methods.interface=test
+
+# comma separated list of available payment method IDs
+methods.available=test-methodid
+
+# config for each method
+methods.test-methodid.method=test-methodid
```

Comment:
I like the idea of being able to convert JSON to configuration, but I wouldn't pass JSON to the Payone adapter directly. I would still have a PayoneConfiguration class with the required fields and then a factory or something that expects JSON and creates a PayoneConfiguration from it.